lib/promoted/ruby/client/options.rb

```diff
@@ -177,8 +177,8 @@ def compact_insertions
           if limit
             insertions_to_compact = insertions_to_compact[0..limit-1]
           end
-          offset = request[:paging][:offset]
-          offset ||= 0
+          paging = request[:paging] || {}
+          offset = paging[:offset].to_i
```

Comment:
are the insertions here the request insertions or response insertions?

Reply:
It should just be ResponseInsertions but the code logs both.  There are messages about this in the ruby slack channel.